### PR TITLE
Adds DoT tracking to Affliction Warlock

### PR DESCRIPTION
**Pull request by LucasCH13**
Added DoT Tracking + Bar Text Variables for Agony, Corruption, Unstable Affliction, Vile Taint/Phantom Singularity, Soul Rot and Haunt

Note for Corruption : There is a talent for Affliction called Absolute Corruption that makes Corruption duration infinite so i don't know how you would make this bar text with infinite value in it for now it shows 0 😄 

Let me know if i missed anything 😃

### CODE REVIEW

**Comment by Twintop** [comment]
Pulling this down to have a play with it. I'll leave some comments inline in addition to this comment.

A few things that need to be added:
- DoT count variables (excluding Phanttom Singularity)
- DoT up/down/pandemic colors config in `Font & Text`.

I see you've got code copypasta's for the colors config already, which is good. I can get these two things added into this branch.

> Note for Corruption : There is a talent for Affliction called Absolute Corruption that makes Corruption duration infinite so i don't know how you would make this bar text with infinite value in it for now it shows 0 😄

There's a couple considerations here:
- Have the bar text (`lookupLogic["$corruptionTime"]`) validity always return `true` *️⃣ 
- Have the bar text to be displayed (`lookup["$corruptionTime"]`) either be an empty string, or, ∞ infinity (this is what the bar would do in Shadowlands when you had enough Anima Powers in Torghast to never, theoretically, lose Void Form)
- *️⃣ = There needs to be an exception added for when you're currently targeting a player, since the duration is 24sec then
- Related to the last, we also need a separate pandemic time check when targeting a player.

File: ClassModules/Warlock.lua [hunk 1]
```diff
@@ -494,10 +700,21 @@ local function SwitchSpec()
 		
 		---@type TRB.Classes.TargetData
 		TRB.Data.snapshotData.targetData = TRB.Classes.TargetData:New()
+		local spellsData = TRB.Data.spellsData --[[@as TRB.Classes.SpellsData]]
+		local targetData = TRB.Data.snapshotData.targetData
+		local spells = spellsData.spells --[[@as TRB.Classes.Priest.ShadowSpells]]
```
Comment (Twintop) [inline ClassModules/Warlock.lua]:
```suggestion
		local spells = spellsData.spells --[[@as TRB.Classes.Warlock.AfflictionSpells]]
```

**Comment by Twintop** [comment]
I've added the DoT count variables where appropriate (not for Unstable Affliction, Soul Rot, or Phantom Singularity). I've also added the DoT coloring options. It was working without it explicitly available for Affliction because of how Global Bar settings and backfilling of common settings worked.

I also changed a couple variable names to match the formatting. Some of your variables might be broken in your bar now (e.g. `$soulrotTime` -> `$soulRotTime`).

**Comment by Twintop** [comment]
I've added Absolute Corruption support, but haven't been able to test it in PvP (though it *should* work). If that's working OK, then the only things left are all things I can do tomorrow:
- News
- Google Translate localizations (would be great if you can add French localizations for the new entries!)
- Add some default bar text

I don't think Affliction is out of the "Experimental" state but it definitely isn't "Minimal" anymore. :D Should have a look at Hero talents to incorporate. 🤔

**Comment by LucasCH13** [comment]
Thanks for the help ! 

I will try in PvP with a friend to see if it works correctly and also translate all entries (including those for Balance Druid) as i get home after work 😄

**Pull request merged**